test/svm/SvmSpoke.Fill.AcrossPlus.ts

```diff
@@ -0,0 +1,355 @@
+    const [fillStatusPDA] = PublicKey.findProgramAddressSync(
+      [Buffer.from("fills"), relayHashUint8Array],
+      program.programId
+    );
+
+    accounts = {
+      state,
+      signer: relayer.publicKey,
+      mintAccount: mint,
+      relayerTokenAccount: relayerATA,
+      recipientTokenAccount: handlerATA,
+      fillStatus: fillStatusPDA,
+      tokenProgram: TOKEN_PROGRAM_ID,
+      associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
+      systemProgram: anchor.web3.SystemProgram.programId,
+    };
+  }
+
+  async function createApproveAndFillIx(multicallHandlerCoder: MulticallHandlerCoder) {
+    // Delegate state PDA to pull relayer tokens.
+    const approveIx = await createApproveCheckedInstruction(
+      accounts.relayerTokenAccount,
+      accounts.mintAccount,
+      accounts.state,
+      accounts.signer,
+      BigInt(relayAmount),
+      mintDecimals
+    );
+
+    const remainingAccounts: AccountMeta[] = [
+      { pubkey: handlerProgram.programId, isSigner: false, isWritable: false },
+      ...multicallHandlerCoder.compiledKeyMetas,
+    ];
+
+    const relayHash = Array.from(calculateRelayHashUint8Array(relayData, chainId));
+
+    // Prepare fill instruction.
+    const fillIx = await program.methods
+      .fillV3Relay(relayHash, relayData, new BN(1), relayer.publicKey)
+      .accounts(accounts)
+      .remainingAccounts(remainingAccounts)
+      .instruction();
+
+    return { approveIx, fillIx };
+  }
+
+  before("Creates token mint and associated token accounts", async () => {
+    mint = await createMint(connection, payer, owner, owner, mintDecimals);
+    relayerATA = (await getOrCreateAssociatedTokenAccount(connection, payer, mint, relayer.publicKey)).address;
+
+    await mintTo(connection, payer, mint, relayerATA, owner, seedBalance);
+
+    await connection.requestAirdrop(relayer.publicKey, 10_000_000_000); // 10 SOL
+
+    [handlerSigner] = PublicKey.findProgramAddressSync([Buffer.from("handler_signer")], handlerProgram.programId);
+    handlerATA = (await getOrCreateAssociatedTokenAccount(connection, payer, mint, handlerSigner, true)).address;
+  });
+
+  beforeEach(async () => {
+    finalRecipient = Keypair.generate().publicKey;
+    finalRecipientATA = (await getOrCreateAssociatedTokenAccount(connection, payer, mint, finalRecipient)).address;
+
+    state = await initializeState();
+
+    const initialRelayData = {
+      depositor: finalRecipient,
+      recipient: handlerSigner, // Handler PDA that can forward tokens as needed within the message call.
+      exclusiveRelayer: relayer.publicKey,
+      inputToken: mint, // This is lazy. it should be an encoded token from a separate domain most likely.
+      outputToken: mint,
+      inputAmount: new BN(relayAmount),
+      outputAmount: new BN(relayAmount),
+      originChainId: new BN(1),
+      depositId: new BN(Math.floor(Math.random() * 1000000)), // force that we always have a new deposit id.
```

Comment:
note we are going to have merge conflicts & breaks between your PR and this https://github.com/across-protocol/contracts/pull/759/files due to `depositId` and how it's defined changing some.